app/services/experiment_control_service.rb

```diff
@@ -0,0 +1,94 @@
+class InvalidExperiment < StandardError; end
+
+class ExperimentControlService
+  class << self
+    def start_current_patient_experiment(name, days_til_start, days_til_end)
+      raise ArgumentError, "Start date must be before end date" if days_til_end < days_til_start
+
+      existing_experiment = Experimentation::Experiment.where(experiment_type: "current_patient_reminder", state: ["selecting", "live"])
+      raise InvalidExperiment, "A current patient experiment is already in progress" if existing_experiment.any?
+
+      experiment = Experimentation::Experiment.find_by!(name: name, experiment_type: "current_patient_reminder")
+      experiment_start = days_til_start.days.from_now.beginning_of_day
+      experiment_end = days_til_end.days.from_now.end_of_day
+
+      experiment.state_selecting!
```

Comment:
something like `start_selecting!` would be more clear here I think. 

We also may want to bring in a proper state machine gem, let's zoom about it.

Reply:
If I'm adding the date validations to the model, I will have to update the dates here along with the state to validate the dates. So the name of the method (which is just an enum method) becomes irrelevant, but we still might want a state machine.